src/main/java/view/DashboardView.java

```diff
@@ -81,93 +96,201 @@ public DashboardView(DashboardViewModel dashboardViewModel,
-                            //  to obtain the needed information?
-                        }
-                    }
+                evt -> {
+                    historyViewModel.firePropertyChanged();
+                    viewManagerModel.setActiveView(historyViewModel.getViewName());
+                    viewManagerModel.firePropertyChanged();
                 }
         );
 
-        // NOTE: may or may not be necessary
-        this.setLayout(new BoxLayout(this, BoxLayout.Y_AXIS));
-
         // Make the various display items visible to the user on the display.
-        this.add(title);
-        this.add(welcome);
-        this.add(buttons);
+        this.setLayout(new BoxLayout(this, BoxLayout.X_AXIS));
+        setLayout(new BorderLayout());
+        GridBagConstraints gbc = new GridBagConstraints();
+
+        topPanel = new JPanel(new GridBagLayout());
+        JPanel intermediatePanel1 = new JPanel(new GridBagLayout());
+        middlePanel = new JPanel(new GridBagLayout());
+        JPanel intermediatePanel2 = new JPanel(new GridBagLayout());
+        bottomPanel = new JPanel(new GridBagLayout());
+
+        gbc.gridx = 0;
+        gbc.gridy = 0;
+        gbc.anchor = GridBagConstraints.WEST;
+//        buttons.setPreferredSize(new Dimension(30, 80));
+        topPanel.add(buttons, gbc);
+
+        gbc.gridx = 1;
+        gbc.weightx = 1;
+        gbc.fill = GridBagConstraints.HORIZONTAL;
+        topPanel.add(Box.createHorizontalGlue(), gbc);
+
+        // Spice up the labels a bit.
+        Font labelFont = balanceLabel.getFont();
+//        balanceLabel.setForeground(new Color(14, 20, 138));
+        balanceLabel.setSize(25, 5);
+        balanceLabel.setFont(labelFont.deriveFont(Font.BOLD));
+
+        // Spice up the actual balance (field) as well.
+        Font fieldFont = balanceField.getFont();
+        balanceField.setForeground(Color.DARK_GRAY);
+        balanceField.setFont(fieldFont.deriveFont(Font.ITALIC));
+
+        gbc.gridx = 2;
+        gbc.gridy = 0;
+        gbc.weightx = 0;
+        gbc.fill = GridBagConstraints.NONE;
+        topPanel.add(balanceLabel, gbc);
+
+        gbc.gridx = 3;
+        gbc.gridy = 0;
+        gbc.anchor = GridBagConstraints.EAST;
+        gbc.weightx = 0;
+        topPanel.add(balanceField, gbc);
+
+        // Get DashboardState after running the use case
+        dashboardController.execute();
+        DashboardState state = dashboardViewModel.getState();
+
+        HashMap<String, String> userStats = state.getUserStats();
+        if (userStats != null) {
+//            userStatsTable.setPreferredSize(new Dimension(100, 200));
+            userStatsTable.setModel(new TableModel(userStats));
+        }
+
+        userStatsTable.setName("User Information");
+
+//        userStatsTable.setMinimumSize(new Dimension(100, 100));
+
+        // Create new default table model for displaying stock information.
+        DefaultTableModel tableModel = new DefaultTableModel();
```

Comment:
I am aware that I created the tables in the initializer as opposed to inside of the `propertyChange` method, as it is done in the other Views. 

I figured that because the only time the tables should change is when the use case is called, they should not be in the property change but instead in the initializer. The other views do not have a refresh button like the dashboard does, so I figured this made a difference and thus warranted different implementation.

Reply:
I think its better to have this in the `propertyChange` method, I think its better if 'refresh' automatically once a sell/buy/reset has been performed rather than have the user have to manually do so. I understand that this may require a little more work but I think it would greatly simplify the look of the view and the user experience we provide.

Reply:
If we do this, I reckon we scrap the refresh button.

Reply:
If it is in the initializer, **it only runs once**, regardless of whatever button is pressed.

If it it something that changes after we hit the run code button on IntelliJ, it has to be in the propertyChanged function